drivers/src/dma.rs

```diff
@@ -612,3 +613,55 @@ impl<'a> DmaRecovery<'a> {
         });
     }
 }
+
+pub struct DmaOtpCtrl<'a> {
+    base: AxiAddr,
+    dma: &'a Dma,
+}
+
+impl<'a> DmaOtpCtrl<'a> {
+    #[inline(always)]
+    pub fn new(base: AxiAddr, dma: &'a Dma) -> Self {
+        Self { base, dma }
+    }
+
+    /// Returns a register block that can be used to read
+    /// registers from this peripheral, but cannot write.
+    #[inline(always)]
+    pub fn with_regs<T, F>(&self, f: F) -> CaliptraResult<T>
+    where
+        F: FnOnce(FuseCtrlRegisterBlock<&DmaMmio>) -> T,
+    {
+        let mmio = DmaMmio::new(self.base, self.dma);
+        // SAFETY: we aren't referencing memory directly
+        let regs = unsafe {
+            FuseCtrlRegisterBlock::new_with_mmio(
+                // substract the recovery offset since all recovery registers are relative to it
```

Comment:
nit: comment refers to recovery registers, but should be OTP registers (or just change to "registers")